frontend/apps/reader/modules/readerflipping.lua

```diff
@@ -83,22 +74,34 @@ function ReaderFlipping:onSetStatusLine()
 end
 
 function ReaderFlipping:paintTo(bb, x, y)
-    if self.ui.highlight.select_mode then
-        if self[1][1] ~= self.select_mode_widget then
-            self[1][1] = self.select_mode_widget
-        end
+    local widget
+    if self.ui.paging and self.view.flipping_visible then
+        -- pdf page flipping or bookmark browsing mode
+        local icon_size = Screen:scaleBySize(32)
+        widget = IconWidget:new{
+            icon = self.ui.paging.bookmark_flipping_mode and "bookmark" or "book.opened",
+            width = icon_size,
+            height = icon_size,
+        }
+    elseif self.ui.highlight.select_mode then
+        -- highlight select mode
+        local icon_size = Screen:scaleBySize(36)
+        widget = IconWidget:new{
+            icon = "texture-box",
+            width = icon_size,
+            height = icon_size,
+            alpha = true,
+        }
     elseif self.ui.rolling and self.ui.rolling.rendering_state then
-        local widget = self:getRollingRenderingStateIconWidget()
+        -- epub rerendering
+        widget = self:getRollingRenderingStateIconWidget()
+    end
+    if widget then
         if self[1][1] ~= widget then
             self[1][1] = widget
         end
```

Comment:
And there, as you created new widget, you'll probably always end up replacing it (except may be with the rolling rendering stuff where we reuse them).